distributed/tests/test_preload.py

```diff
@@ -161,61 +155,40 @@ async def test_preload_import_time(cleanup):
         del backends["foo"]
 
 
-class MyHandler(web.RequestHandler):
-    def get(self):
-        self.write(
-            """
-def dask_setup(dask_server):
-    dask_server.foo = 1
-""".strip()
+@gen_test()
+async def test_web_preload():
+    def preload():
+        with server_sock.accept()[0] as client_sock, client_sock.makefile("rwb") as f:
+            assert f.readline() == b"GET /preload HTTP/1.1\r\n"
+            f.write(
+                b"HTTP/1.1 200 OK"
+                b"\r\nContent-Length: 53"
+                b"\r\n\r\n"
+                b"def dask_setup(dask_server):"
+                b"\n    dask_server.foo = 1"
+                b"\n"
+            )
```

Comment:
This feels a bit off. Why are we now writing a binary HTTP response to a plain socket?

Reply:
It used to be spinning up a whole tornado server in a subprocess - which leaked a bunch of stuff

Reply:
I can't use an async server because preload uses a blocking urllib3 request

Reply:
I don't mind us spinning up a whole HTTP server. I consider this very hard to maintain and extremely difficult to extend. If the tornado server is leaking, we should think about how to replace it with something else but relying on plain sockets is not something I like to do for this even though this very particular application is easy enough to understand.
Let's say we'd like to protect the preload route with Auth. How would this test then look like? What if we want the preload route to accept additional parameters? There are a bunch of situations I can see happening that would require us to write much more complicated HTTP. If any other test dependencies can help us simplify this, that's something to consider.

Is this actually required to get pytest-asyncio out? I would rather let us ignore a few resource leakages.
If a blocking urllib3 request is the issue, maybe we should not have anything blocking in, in the first place and we should have a look at this, but at a later time.

Reply:
@fjetter I had a look at vcrpy and decided it was too invasive - so I think a plain `mock.patch` on urllib3 is better